### PR TITLE
CSHARP-30 | @jdwjdwjdw | c-ShARP masthead and hero banner variant

### PR DESCRIPTION
**Pull request by jdwjdwjdw**
# READY FOR REVIEW

# Summary
- [CSHARP-30](https://stanfordits.atlassian.net/browse/CSHARP-30): Implement subtheme header
- This PR includes changes to the masthead - lockup, decoupled menu, main nav, and search.
- This PR also includes a transparent overlay hero banner variant
- See [Figma](https://www.figma.com/file/Qquk8HFUxpilJHjHaDt7jj/cShARP-Subtheme?node-id=2504%3A14421&mode=dev) under `Page 1`. I ave attached screenshots of the most up-to-date section of that figma page, which is the best area to refer to while reviewing.
Screenshot 2024-04-03 at 11 10 07 AM
Screenshot 2024-04-03 at 11 10 24 AM
Screenshot 2024-04-03 at 11 08 35 AM

# Review By (Date)
- ASAP

# Urgency
- High

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Navigate to `/admin/config/system/basic-site-settings`. Under the `Site Settings/Features` tab, check `Use Drop Down Menus`. Under the `Search` tab, check `Hide Site Search`. Save.
3. Navigate to `/admin/config/system/lockup-settings`. Select the lockup `Option O: Line 4`, with Line 4 being `c-ShARP`. Save.
4. Navigate to `/admin/structure/menu/manage/main`. Recreate the menu structure seen at https://csharp-dev.sites.stanford.edu/admin/structure/menu/manage/main. You won't need to add all of the sub-menu items, but make sure the man nav items are the same. Add a few sub-menu items to test those as well. Save.
5. Navigate to the homepage and confirm the the masthead matches figma and looks good.
6. For testing the transparent overlay hero banner variant: Create a basic page with a top banner that matches the banner in the figma docs / screenshots above. Click the Banner `Behavior` tab and select the `Transparent Overlay` Banner style.
7. Confirm the Transparent Overlay banner matches figma at all breakpoints
8. Review code ⛷️ 

# Associated Issues and/or People
- [CSHARP-30](https://stanfordits.atlassian.net/browse/CSHARP-30): Implement subtheme header

[CSHARP-30]: https://stanfordits.atlassian.net/browse/CSHARP-30?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mdyoung3** [comment]
@jdwjdwjdw On step four, do I remove all the old menu items first?

**Comment by jdwjdwjdw** [comment]
> @jdwjdwjdw On step four, do I remove all the old menu items first?

Yeah, or just rename the current ones. Just trying to get it to a similar state as it should look for the final site.

**Comment by mdyoung3** [comment]
@jdwjdwjdw 

Is this the correct font size and width? If so, my local is a bit off from this. 

Screenshot 2024-04-08 at 4 06 18 PM

File: src/js/c_sharp_subtheme.behaviors.js
Comment (mdyoung3) [inline src/js/c_sharp_subtheme.behaviors.js:1]: @jdwjdwjdw Any reason to keep this?

Comment (jdwjdwjdw) [reply src/js/c_sharp_subtheme.behaviors.js:1]: Nope - good catch. I deleted that.

File: c_sharp_subtheme.libraries.yml [hunk 1]
```diff
@@ -87,7 +87,11 @@
 #       css/pattern-library/theme/print.css: { media: print }
 
 allpages:
-  version: VERSION
   css:
     theme:
       dist/css/main.css: { }
+  js:
+    src/js/c_sharp_subtheme.behaviors.js: { }
```
Comment (mdyoung3) [inline c_sharp_subtheme.libraries.yml]: @jdwjdwjdw Remove this as well to prevent warning messages.

**Comment by mdyoung3** [comment]
@jdwjdwjdw There's quite a bit of time before the menu appears on the page (see screencast). IDK if this is tolerated behavior or if loading the script sooner will prevent this. 


https://github.com/SU-SWS/c_sharp_subtheme/assets/8387457/44a5252c-3ed2-4847-b5ed-1a8d0f9de74b

File: c_sharp_subtheme.libraries.yml (hunk 1 above)
Comment (jdwjdwjdw) [reply c_sharp_subtheme.libraries.yml]: I think you're looking at an outdated version of this file - I removed it in my latest commit.

Comment (mdyoung3) [reply c_sharp_subtheme.libraries.yml]: @jdwjdwjdw You're right. I refreshed the page a few times on GitHub to make sure, but my local has it removed. Disregard.

**Comment by jdwjdwjdw** [comment]
@mdyoung3 I think the loading time is something that has been discussed for the decoupled menu on other subthemes, but I'm not sure if that's something we need to worry about here since it appears to be consistent whenever the decoupled menu is used. @pookmish if you have other thoughts, let us know.

**Comment by pookmish** [comment]
Yeah, that's a "local" issue with performance. on Acquia, the service is much more responsive and we shouldn't have such a delay as you are seeing.

**Comment by mdyoung3** [comment]
@jdwjdwjdw It looks like maybe the transparent header is smaller than the figma:

Figma:
Screenshot 2024-04-11 at 1 11 39 PM

My local:
Screenshot 2024-04-11 at 1 11 56 PM

**Comment by pookmish** [comment]
@mdyoung3, did you intend to close this PR?

**Comment by mdyoung3** [comment]
Nope. Re-opened. Thanks

File: decoupled-menu/src/main-menu.island.tsx [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+    background: transparent;
+    height: 100%;
+    width: 100%;
+    box-shadow: none;
+    border-top: none;
+`
+
+const MobileMenuButton = styled.button`
+  position: absolute;
+  top: -70px;
```
Comment (mdyoung3) [inline decoupled-menu/src/main-menu.island.tsx:55]: I'd boost this to -80px. Seems to look better. 

Screenshot 2024-04-11 at 1 52 39 PM

Comment (jdwjdwjdw) [reply decoupled-menu/src/main-menu.island.tsx:55]: That was something the designer specifically requested - they wanted the `Menu` label to align with the bottom of the `c-ShARP` text.

**Comment by jdwjdwjdw** [comment]
> @jdwjdwjdw It looks like maybe the transparent header is smaller than the figma:
> 
> Figma: Screenshot 2024-04-11 at 1 11 39 PM
> 
> My local: Screenshot 2024-04-11 at 1 11 56 PM

Looks like my heading styles are now being overwritten due to recent Stanford Profile helper changes at https://github.com/SU-SWS/stanford_profile_helper/blob/9.x/modules/stanford_profile_styles/lib/scss/component/paragraph/card/_card.scss. I'll update these styles to fix.

**Comment by mdyoung3** [comment]
@jdwjdwjdw It's looking correct now from the U.I. I'd say this is good to go. I'm not strong with TypeScript. But I gave a look-see of the code.

**Review by mdyoung3 — APPROVED** [review]

**Pull request merged**